libs/zephyr-agent/src/lib/node-persist/token.ts

```diff
@@ -1,36 +1,122 @@
 import { clear, getItem, init, removeItem, setItem } from 'node-persist';
 import { homedir } from 'node:os';
 import { join } from 'node:path';
+import { existsSync, mkdirSync } from 'node:fs';
 import { StorageKeys, ZE_PATH } from './storage-keys';
 import { getSecretToken } from './secret-token';
+import { ze_log } from '../logging';
+import * as lockfile from 'proper-lockfile';
 
+// Storage directory path
+const storageDir = join(homedir(), ZE_PATH);
+
+// Ensure the directory exists
+if (!existsSync(storageDir)) {
+  mkdirSync(storageDir, { recursive: true });
+}
+
+// Path for the lockfile
+const tokenLockPath = join(storageDir, `${StorageKeys.ze_auth_token}.lock`);
+
+// Make storage initialization lazy and synchronized
 const storage = init({
-  dir: join(homedir(), ZE_PATH),
+  dir: storageDir,
 });
 
+/** Save token with file locking to prevent race conditions between processes */
 export async function saveToken(token: string): Promise<void> {
-  await storage;
-  await setItem(StorageKeys.ze_auth_token, token);
+  let release;
+  try {
+    // Acquire a lock before writing the token
+    // stale: 10000 (10s) - consider a lock stale after 10s (if process crashed)
+    // retries: 5 - attempt to acquire the lock 5 times
+    // retryWait: 500 - wait 500ms between retries
+    release = await lockfile.lock(tokenLockPath, {
+      stale: 10000,
+      retries: 5,
+    });
+
+    // Once we have the lock, save the token
+    await storage;
+    await setItem(StorageKeys.ze_auth_token, token);
+
+    ze_log('saveToken: Token saved successfully with lock');
```

Comment:
```suggestion
```

Let's log only useful stuff, too much log usually hides or at least make it harder to find the actual problem between a thousand logs